mdoc/Mono.Documentation/MDocUpdater.cs

```diff
@@ -2518,25 +2518,80 @@ private void UpdateMember (DocsNodeInfo info, FrameworkTypeEntry typeEntry, Dict
-
-            AddXmlNode(
-                xmlElement.SelectNodes (elementXPath).Cast<XmlElement> ().ToArray (),
-                valueMatches,
-                setValue,
-                () => makeNewNode(true),
-                type);
+        private static void UpdateSignature(MemberFormatter formatter, TypeDefinition type, XmlElement xmlElement, FrameworkTypeEntry typeEntry)
+        {
+            string elementName = "TypeSignature";
+            string elementXPath = $"{elementName}[@Language='" + formatter.Language + "']";
+            var existingElements = QueryXmlElementsByXpath(xmlElement, elementXPath).ToList();
+
+            if (typeEntry.TimesProcessed > 1 && existingElements.Any())
+                return;
+
+            // if first framework of type, clear signatures and generate from scratch
+            if (typeEntry.Framework.IsFirstFrameworkForType(typeEntry))
+            {
+                existingElements.ForEach(e => e.ParentNode.RemoveChild(e));
+                existingElements = QueryXmlElementsByXpath(xmlElement, elementXPath).ToList();
+            }
+
+            // start to process 
+            var usageSample = formatter.UsageFormatter?.GetDeclaration(type);
+            var valueToUse = formatter.GetDeclaration(type);
+            if (valueToUse == null && usageSample == null)
+                return;
+
+            bool elementFound = false;
+
+            // if there is already signature found under same formatter, update fxa
+            if (existingElements.Any())
+            {
+                foreach (var element in existingElements)
+                {
+                    var val = element.GetAttribute("Value");
+                    var usage = element.GetAttribute("Usage");
+                    if (val == (valueToUse ?? "") && usage == (usageSample ?? ""))
+                    {
+                        // add fxa
+                        string newfxa = FXUtils.AddFXToList(element.GetAttribute(Consts.FrameworkAlternate), typeEntry.Framework.Name);
+                        element.SetAttribute(Consts.FrameworkAlternate, newfxa);
+                        elementFound = true;
+                    }
+                }
+            }
+
+            if (!elementFound) //not exists, add signature with fxa
+            {
+                var newElement = xmlElement.OwnerDocument.CreateElement(elementName);
+                xmlElement.AppendChild(newElement);
+                newElement.SetAttribute("Language", formatter.Language);
+
+                if (!string.IsNullOrWhiteSpace(valueToUse))
+                    newElement.SetAttribute("Value", valueToUse);
+
+                if (!string.IsNullOrWhiteSpace(usageSample))
+                    newElement.SetAttribute("Usage", usageSample);
+
+                newElement.SetAttribute(Consts.FrameworkAlternate, typeEntry.Framework.Name);
+            }
+
+            // if last framework for type, check if need to purge FrameworkAlternate attribute

```

Comment:
@joelmartinez, I will create a separate pr to move method `PurgeFrameworkAlternateAttribute` to DocUtils.cs, change this section of code to use generic method.

Reply:
ok